### PR TITLE
major changes in cli.py

### PR DESCRIPTION
**Pull request by makefu**
- make py3 compatible
- make url configurable with the --url parameter
- input file is given via argument, not parameter
- return url now contains host+url
- up version to 0.2
- prepare NamedTemporaryFile for stdin, but guess_type does not seem to work

### CODE REVIEW

File: bepasty_cli/cli.py [hunk 1]
```diff
@@ -5,22 +6,25 @@
...
 from mimetypes import guess_type
-#from __future__ import print_function
 import requests
 
+#from tempfile import NamedTemporaryFile
+
 import click
 
 @click.command()
+@click.argument('fileobj',nargs=1,required=False)
```
Comment (NotANormalNerd) [inline bepasty_cli/cli.py]: Please run it through a PEP8 test and improve accordingly

File: bepasty_cli/cli.py [hunk 2]
```diff
@@ -30,15 +34,26 @@ def main(token, fileobj, fname, ftype):
         stdin = False
 
     else:
+        # tried to write stdin in tempfile and run guess_file, but guess_file
+        # only seem to look on the file extension ...
+        #tmpfile = NamedTemporaryFile(delete=False)
```
Comment (NotANormalNerd) [inline bepasty_cli/cli.py]: No commented code. Either have it or remove.

File: bepasty_cli/cli.py [hunk 3]
```diff
@@ -30,15 +34,26 @@ def main(token, fileobj, fname, ftype):
...
         stdin = True
 
     if not ftype:
+        # guess_type sucks buttocks, better use python-magic?
         ftype, enc = guess_type(fname)
+        print('guessed filetype: {}'.format(ftype))
         if not ftype:
-            ftype = 'application/octet-stream'
+            #ftype = 'application/octet-stream'
+            ftype = 'text/plain'
```
Comment (NotANormalNerd) [inline bepasty_cli/cli.py]: Default bepasty file-type for unkown files is 'application/octet-stream'

**Comment by makefu** [comment]
well, maybe bepasty default file-type is not the right one for a pastebin?

Imho what one expects when running `echo "lol text" | bepasty-cli` is that you can actually see the text on the the pastebin, not needing to 'click to download binary'.

so my suggestion is either fix mimetype for stdin by writing a temporary file and analyze it with magic or keep default filetyp text/plain.

If none is happening i am bound to maintain a fork with this single line changed ;P

**Comment by NotANormalNerd** [comment]
You can always do `echo "lol text" | bepasty-cli -t txt`

**Comment by makefu** [comment]
maybe you have meant `echo "lol text" | bepasty-cli -t text/plain` because `txt` or `text` simply is no mimetype. This is about usability, not about what is *according to the specs(tm)*

**Pull request merged**